executor/executor.go

```diff
@@ -398,6 +382,7 @@ func (e *CheckTableExec) run(ctx context.Context) error {
 			}
 		}
 	}
+	e.done = true
```

Comment:
This is different from the current behaviour. You can use defer to set e.done to true after line 370. So e.done will be set to true whether meet error or not.